From: Sidhartha <sidhartha.kumar@oracle.com>
Subject: maple_tree: remove conditionals to detect wr_node_store
Date: Wed, 9 Oct 2024 11:20:07 -0400

In mas_wr_store_type(), we check if new_end < mt_slots[wr_mas->type].  If
this check fails, we know that ,after this, new_end is >= mt_min_slots. 
Checking this again when we detect a wr_node_store later in the function
is reduntant.  Because this check is part of an OR statement, the
statement will always evaluate to true, therefore we can just get rid of
it.

Link: https://lkml.kernel.org/r/20241009152007.2096-1-sidhartha.kumar@oracle.com
Signed-off-by: Sidhartha Kumar <sidhartha.kumar@oracle.com>
Suggested-by; Wei Yang <richard.weiyang@gmail.com>
Cc: "Liam R. Howlett" <Liam.Howlett@oracle.com>
Cc: Matthew Wilcox <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/maple_tree.c |    9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

--- a/lib/maple_tree.c~maple_tree-remove-conditionals-to-detect-wr_node_store
+++ a/lib/maple_tree.c
@@ -4252,14 +4252,7 @@ static inline void mas_wr_store_type(str
 		return;
 	}
 
-	if (mte_is_root(mas->node) || (new_end >= mt_min_slots[wr_mas->type]) ||
-		(mas->mas_flags & MA_STATE_BULK)) {
-		mas->store_type = wr_node_store;
-		return;
-	}
-
-	mas->store_type = wr_invalid;
-	MAS_WARN_ON(mas, 1);
+	mas->store_type = wr_node_store;
 }
 
 /**
_
